src/o365/graph/commands/teams/teams-clone.ts

```diff
@@ -0,0 +1,208 @@
+import * as request from 'request-promise-native';
+import auth from '../../GraphAuth';
+import Utils from '../../../../Utils';
+import config from '../../../../config';
+import commands from '../../commands';
+import GlobalOptions from '../../../../GlobalOptions';
+import { CommandOption, CommandValidate } from '../../../../Command';
+import GraphCommand from '../../GraphCommand';
+
+const vorpal: Vorpal = require('../../../../vorpal-init');
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  teamId: string;
+  displayName: string;
+  mailNickname: string;
+  partsToClone: string;
+  description?: string;
+  classification?: string;
+  visibility?: string;
+}
+
+class GraphTeamsCloneCommand extends GraphCommand {
+  public get name(): string {
+    return `${commands.TEAMS_CLONE}`;
+  }
+
+  public get description(): string {
+    return 'Creates a clone of a Microsoft Teams team';
+  }
+
+  public getTelemetryProperties(args: CommandArgs): any {
+    const telemetryProps: any = super.getTelemetryProperties(args);
+    telemetryProps.description = typeof args.options.description !== 'undefined';
+    telemetryProps.classification = typeof args.options.classification !== 'undefined';
+    telemetryProps.visibility = typeof args.options.visibility !== 'undefined';
+    return telemetryProps;
+  }
+
+  public commandAction(cmd: CommandInstance, args: CommandArgs, cb: () => void): void {
+    auth
+      .ensureAccessToken(auth.service.resource, cmd, this.debug)
+      .then((): request.RequestPromise => {
+
+        const requestOptions: any = {
+          url: `${auth.service.resource}/v1.0/teams/${encodeURIComponent(args.options.teamId)}/clone`,
+          headers: Utils.getRequestHeaders({
+            authorization: `Bearer ${auth.service.accessToken}`,
+            "content-type": "application/json",
+            accept: 'application/json;odata.metadata=none'
+          }),
+          json: true,
+          body: {
+            displayName : args.options.displayName,
+            mailNickname : args.options.mailNickname,
```

Comment:
Could you please auto generate mailNickname to be send with the body? For example if the user specifies for displayName "My cool team", we can escape everything that is not number or char (a-zA-Z0-9) so the nickname can look like "Mycoolteam" then send that as mailNickname with the payload. 

See: https://github.com/pnp/office365-cli/pull/939#discussion_r279938818

Reply:
@VelinGeorgiev, I have updated the parameter with a static value "TBD" as we not sure about the validations performed on this. If required, please feel free to change.